Tasks/DotnetCLIRestore/restorecommand.ts

```diff
@@ -0,0 +1,270 @@
+            false /* useNugetToModifyConfigFile */);
+
+        let credCleanup = () => { return; };
+
+        const sources: Array<auth.IPackageSource> = new Array<auth.IPackageSource>();
+        let feedUri: string = "https://nuget.pkg.github.com/" + username + "/index.json";
+        sources.push(<auth.IPackageSource>
+            {
+                feedName: "github",
+                feedUri: feedUri,
+                isInternal: false
+            });
+
+        // Creating NuGet.config for the user
+        if (sources.length > 0) {
+            tl.debug(`Adding the following sources to the config file: ${sources.map(x => x.feedName).join(';')}`);
+            nuGetConfigHelper.addSourcesToTempNuGetConfig(sources);
+            credCleanup = () => {
+                tl.rmRF(nuGetConfigHelper.tempNugetConfigPath);
+            };
+            nuGetConfigPath = nuGetConfigHelper.tempNugetConfigPath;
+        } else {
+            tl.debug('No sources were added to the temp NuGet.config file');
+        }
+
+        // Setting creds in the temp NuGet.config if needed
+        nuGetConfigHelper.setAuthForSourcesInTempNuGetConfig();
+        const configFile = nuGetConfigHelper.tempNugetConfigPath;
+        nuGetConfigHelper.backupExistingRootNuGetFiles();
+        const dotnetPath = tl.which('dotnet', true);
+
+        let projectFiles = CreateProjectFiles(packageName, packageVersion);
+
+        try {
+            for (const projectFile of projectFiles) {
+                await dotNetRestoreAsync(dotnetPath, projectFile, packagesDirectory, packageVersion, configFile, noCache, verbosity);
+            }
+        } finally {
+            credCleanup();
+            nuGetConfigHelper.restoreBackupRootNuGetFiles();
+        }
+
+        tl.setResult(tl.TaskResult.Succeeded, tl.loc('PackagesInstalledSuccessfully'));
+
+    } catch (err) {
+        tl.error(err);
+        if (buildIdentityDisplayName || buildIdentityAccount) {
+            tl.warning(tl.loc('BuildIdentityPermissionsHint', buildIdentityDisplayName, buildIdentityAccount));
+        }
+
+        tl.setResult(tl.TaskResult.Failed, tl.loc('PackagesFailedToInstall'));
+    }
+}
+
+function CreateProjectFiles(packageName: string, version: string): string[] {
+    let projectFilePath: string = getProjectFilePath();
+    tl.writeFile(projectFilePath, "<Project Sdk=\"Microsoft.NET.Sdk\" />");
+
+    updateXmlFile(projectFilePath, (xml: any): any => {
+        if (xml) {
+            if (xml.getName().toLowerCase() !== "project") {
+                throw Error("Expected project element");
+            }
+            let xmlPropertyGroup = getOrCreateLastElement(xml, "PropertyGroup");
+            let xmlOutputType = xmlPropertyGroup.c("OutputType");
+            let xmlTargetFramework = xmlPropertyGroup.c("TargetFramework");
+            let xmlOutputTypeTxt = xmlOutputType.t("Exe");
+            let xmlTargetFrameworkTxt = xmlTargetFramework.t("netcoreapp2.1");
```

Comment:
Given that now 3.1 is the latest LTS version, should we rather use that ?

Reply:
I tried using 3.0 but was getting some error. So, I will stick to this.